src/mcedit2/editorcommands/analyze.py

```diff
@@ -0,0 +1,146 @@
+        blockArrayHeaders = ['Name', 'ID', 'Data', 'Count']
+        self.setupTable(self.blockArrayData, blockArrayHeaders, blockTableView)
+        
+        
+        entityTableView = self.entityOutputTableView
+        self.entityArrayData = []
+        for c in entityCount, tileEntityCount:
+            for (id, count) in sorted(c.items()):
+                self.entityArrayData.append((id, count,))
+        entityArrayHeaders = ['Name', 'Count']
+        self.setupTable(self.entityArrayData, entityArrayHeaders, entityTableView)
+
+    def setupTable(self, arraydata, headerdata, tableView):
+        tableModel = CustomTableModel(arraydata, headerdata)
+        
+        tableView.setModel(tableModel)
+        tableView.verticalHeader().setVisible(False)
+        tableView.horizontalHeader().setStretchLastSection(True)
+        tableView.resizeColumnsToContents()
+        tableView.resizeRowsToContents()        
+        tableView.setSortingEnabled(True)
+        
+        
+        # -- Exporting stuff --
+        
+    def export_csv(self):
+        startingDir = getUserFilesDirectory()
+        name = self.worldName + "_" + arrow.now().format('DD_MM_YYYY_HH_mm_ss')
+        file = QtGui.QFileDialog.getSaveFileName(QtGui.qApp.mainWindow,
+                                                   self.tr("Export as .csv"),
+                                                   startingDir + "\\" + name,
+                                                   "Comma Seperated Values (*.csv);;Semicolon Seperated Values (*.csv)")
+        if file and file[0]:
+            """
+            Depending on your region, your OS uses ";" or "," as a seperator in .csv files. 
+            (Some countries write 0.5 as 0,5; so they use ; to seperate values).
+            If the user selects Semicolon Seperated Values, we seperate with ";" instead of ","
+            """
+            sep = (";" if (file[1] == "Semicolon Seperated Values (*.csv)") else ",")
+            self.writeFile(file[0], sep)
+           
+    def export_txt(self):
+        startingDir = getUserFilesDirectory()
+        name = self.worldName + "_" + arrow.now().format('DD_MM_YYYY_HH_mm_ss')
+        file = QtGui.QFileDialog.getSaveFileName(QtGui.qApp.mainWindow,
+                                                   self.tr("Export as .txt"),
+                                                   startingDir + "\\" + name,
+                                                   "Text File (*.txt)")
+        if file and file[0]:
+            sep = "\t"
+            self.writeFile(file[0], sep)
+        
+    def writeFile(self, filename, sep):
+        with open(filename, 'w') as f:
+            f.write("Blocks:\n")
+            f.write("Name" + sep + "Id" + sep + "Data" + sep + "Count\n")
+            for b in self.blockArrayData:
+                f.write(unicode(b[0]) + sep + unicode(b[1]) + sep + unicode(b[2]) + sep + unicode(b[3]) + "\n") #xxx Unrolled loop
```

Comment:
Regarding `unicode()`:

Without passing an encoding parameter, it will decode as ASCII. Furthermore, if `b[0]` is already a `unicode`, it will encode as ASCII and then decode as ASCII(!). And on top of that, when you write a `unicode` into an open file that has no encoding set, it will encode right back into ASCII. Every ASCII encoding/decoding is guaranteed to raise encoding errors for foreign characters.

Do you know the data types of the data you have in blockArrayData? Block names should already be `unicode`, since if they contain foreign characters they will be useless for display purposes otherwise. The other three fields are integers.

An open file can only write `str` types without error - if you pass a `unicode` type, it will implicitly encode it to `str` using the ASCII codec. If your intent is to write an UTF-8 encoded file (probably the correct choice), there are a few options:

One is to explicitly encode all `unicode` types using UTF-8 before writing to the file. Without a codec assigned to the file, it can only write `str` types and will implicitly encode `unicode`s as ASCII.

Another is to use `codecs.open` to open the file with a UTF-8 encoder assigned. In this case, you may only write `unicode` types to the file - any `str` types will be implicitly decoded to `unicode` as ASCII before being re-encoded to `str` as UTF-8.

This code will not fail due to encoding errors at the moment because all block names use only ASCII characters, but it will fail once we begin fetching localized block names from Minecraft's translation files or elsewhere.